app/parse.py

```diff
@@ -15,8 +30,102 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+class WebScraper:
+    def __init__(self) -> None:
+        self.options = Options()
+        self.options.add_argument("--headless")
+        self.driver = webdriver.Chrome(options=self.options)
+
+    def click_accept_cookies(self, url: str) -> None:
+        self.driver.get(url)
+        cookies_accept_button = self.driver.find_element(
+            By.CLASS_NAME, "acceptCookies"
+        )
+        if cookies_accept_button.is_displayed():
+            cookies_accept_button.click()
+
+    def scroll_down_page_with_more_button(self, url: str) -> None:
+        self.driver.get(url)
+
+        while True:
+            try:
+                more_button = self.driver.find_element(
+                    By.CSS_SELECTOR, "[class*=scroll-more]"
+                )
+                if more_button.is_displayed():
+                    more_button.click()
+                    time.sleep(1)
+            finally:
+                break
+
+    def scrape_single_page(self, url: str) -> [Product]:
+        result = []
+        self.driver.get(url)
+        products = self.driver.find_elements(
+            By.CSS_SELECTOR, "[class*=product-wrapper]"
+        )
+        for product in products:
+            result.append(
+                Product(
+                    title=product.find_element(
+                        By.CSS_SELECTOR, "a.title"
+                    ).get_attribute("textContent"),
+                    description=product.find_element(
+                        By.CSS_SELECTOR, "[class*=description]"
+                    ).text,
+                    price=float(
+                        product.find_element(
+                            By.CSS_SELECTOR, "[class*=price]"
+                        ).text.replace("$", "")
+                    ),
+                    rating=len(
+                        product.find_elements(
+                            By.CSS_SELECTOR, "p > span.ws-icon-star"
+                        )
+                    ),
+                    num_of_reviews=int(
+                        product.find_element(
+                            By.CSS_SELECTOR, "[class*=review-count]"
+                        ).text.split()[0]
+                    ),
```

Comment:
It's better to split it on separate methods to get each field in a correct format because if product has 20 fields, this method will be overwhelmed